hw/xbox/mcpx_apu.c

```diff
@@ -473,6 +491,87 @@ static void ep_scratch_rw(void *opaque,
                       ptr, addr, len, dir);
 }
 
+static void circular_scatter_gather_rw(MCPXAPUState *d,
+                                       hwaddr sge_base, unsigned int max_sge,
+                                       uint8_t *ptr,
+                                       uint32_t base, uint32_t end,
+                                       uint32_t *cur,
+                                       size_t len,
+                                       bool dir)
+{
+    while (len > 0) {
+        unsigned int bytes_to_copy = end - *cur;
```

Comment:
nit: Should be size_t

I decided to use `unsigned int` because I've also used that in the `scatter_gather_rw` (which is similar). I think we should do a larger cleanup in the future, `unsigned int` is fine for now and doesn't raise the question: "why is it size_t here, but not there?".